torch/linalg/__init__.py

```diff
@@ -2042,3 +2042,25 @@
     >>> torch.dist(Q.transpose(-2, -1) @ Q, torch.eye(4))
     tensor(6.2158e-07)
 """)
+
+trace = _add_docstr(_linalg.linalg_trace, r"""
+trace(input, offset=0) -> Tensor
+
+Returns the sum of the elements of the diagonal. If `offset` is given, the returned value is the sum
```

Comment:
"Computes the trace of a matrix or a batch of traces for a batch of matrices. The trace of a matrix is the sum of elements along its diagonal..."

Then this needs to explain what the diagonal is and how offset works. A proper mathematical formula would help. An issue with the current description is that `i` is not well defined. We may also want to link the Wikipedia article, which does a good job explaining what the diagonal is (https://en.wikipedia.org/wiki/Main_diagonal).

Reply:
To keep the linalg docs with a homogeneous notation, I'm proposing writing something like:

> trace(input, *, offset=0) -> Tensor
> Computes the trace of a square matrix.
>
> Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
> the **trace ** of a square matrix
> :math:`A \in \mathbb{K}^{n \times n}` is defined as
>
> .. math::
>
>     \operatorname{tr}(A) = \sum_{i=0}^{n-1} A_{i,i}
>
> If :attr:`offset` is provided, the sum will run over the elements :math:`A_{i, j}`
> with :math:`i - j = \texttt{offset}`.
>
> Supports input of float, double, cfloat and cdouble dtypes.
> Also supports batches of matrices, and if :attr:`A` is a batch of matrices then
> the output has the same batch dimensions.
>
> Args:
>     A (Tensor): tensor of shape `(*, n, n)` where `*` is zero or more batch dimensions.
>
> Keyword args:
>     offset (int, optional): offset of the diagonal from the main diagonal. A positive value indicates
>                             an upper diagonal, and negative value a lower diagonal. Default: `0`.